### DISCUSSION TITLE
Dynamic UI, blender like

### DISCUSSION THREAD

**Discussion by PladsElsker** [Ozu-Beatmap-Toolset/ozu-gui#21] [Investigation]
I want to make a UI that's as dynamic as blender! 

So basically, you can remodel the interface as much as you want. Widget madness. 

To start off, it would be great to be able to resize the widgets by dragging the edges of divs.

To do that, we could use this package, it looks exactly like what we need:
https://www.npmjs.com/package/vue-resizable

**Comment by PladsElsker** [comment]
Further in dev, we can try to add a dynamic widget handler.

Basically, if the user wants to have 2 playfields open at the same time in parallel, he's going to be able to. 
Modifying 1 playfield updates the other in real time too.

**Comment by PladsElsker** [comment]
I think that the solutions that use CSS are usually really elegant and help solve problems faster.

This is the idea I have for now:
1) The background + the image are taking the whole space, we should add another transparent container div in front to place the different widgets inside;
2) Once that's done, we can add 2 types of nested widgets inside that div: "HorizontalWidget" and "VerticalWidget". 
3) Each widget type is positionned with "relative" in css. The horizontal one has "height: 100%" so that it fills as much as possible the parent container. The vertical div is the same, but with the "width" attribute.
4) We can then place in HTML any widget that we want in every nested box. Each concrete widget is positionned as absolute.
5) We might have to use JS as well to do some tricks that would otherwise be really hard to do in CSS.

To sum up:
- use CSS to easily sort in a tree the area optimally,
- use JS to implement the rest.
